Source/Editor/EditorCore/ResourceOperations.cpp

```diff
@@ -254,6 +254,50 @@ Resource* DuplicateResource(Resource* resource, StringParam expectedNewName)
   return duplicate;
 }
 
+bool MoveResource(Resource* resource, ContentLibrary* targetContentLibrary, ResourceLibrary* targetResourceLibrary)
+{
+    if (PL::gEngine->IsReadOnly())
+    {
+        DoNotifyWarning("Resources", "Cannot rename resources while in read-only mode");
+        return false;
+    }
+
+    ResourceManager* resourceManager = resource->GetManager();
+    String resourceTypeName = resourceManager->mResourceTypeName;
+    ContentItem* contentItem = resource->mContentItem;
+    String previousLocation = contentItem->mLibrary->SourcePath;
+    HandleOf<Resource> resourceHandle = resource;
+
+    ReturnIf(!contentItem->mResourceIsContentItem, false, "Cannot move resource. The resource is either corrupted or a resource that is unable to move.");
+    ReturnIf(targetContentLibrary == nullptr, false, "Cannot move resource to a null library.");
+    ReturnIf(targetContentLibrary->GetReadOnly(), false, "Cannot move resource to a library that is read only.");
+
+    bool result = PL::gContentSystem->MoveContentItem(contentItem, targetContentLibrary);
+    if (result == false)
+    {
+        DoNotifyWarning("Resources", "Moving resource failed.");
+        return result;
+    }
+
+    resource->mResourceLibrary->Resources.EraseValueError(resourceHandle);
+    resource->mResourceLibrary = targetResourceLibrary;
+    targetResourceLibrary->Resources.PushBack(resourceHandle);
+    resource->UpdateContentItem(contentItem);
+    contentItem->SaveContent();
+
```

Comment:
Something that _I think_ is missing here is updating the dependencies on both the source and destination ResourceLibraries.

Reply:
So the great thing about how the content system operates and resolves dependencies, everything has a unique key that anything can lookup. The only thing this may break is plasma fragments for shaders, so this will definitely be something we need to look into when updating the whole dependency system.

Reply:
Right, but the actual Dependencies and Dependents arrays on the ResourceLibrary aren't updated here. It looks like those arrays are important for several things, including shader and script compilation. So if I move a script that other scripts depend on, then I won't be able to unload the old ResourceLibrary, but I will be able to unload the new ResourceLibrary.
Let me know if I'm off base here, but it seems like the only time we setup the Dependencies array is on the initial `LoadPackage` call, which isn't run again after moving resources between libraries.